sonic_sfp/sff8436.py

```diff
@@ -1229,3 +1229,472 @@ def get_data(self):
 
     def get_data_pretty(self):
         return sffbase.get_data_pretty(self, self.dom_data)
+
+class sff8436DomThreshold(sffbase):
```

Comment:
I would suggest adding new APIs to current sff8436Dom class instead of having this new class.
The benefit I can see could be:

1. reduce the duplicated code, there is no reason to have two copies of 'cal_temperature', 'calc_voltage', etc.

2. The threshold is part of the DOM, add new APIs would be more intuitional.

Reply:
Hi,

Addressed the review comment. Removed the duplicate code and added new APIs in sff8436Dom